src/workerd/api/node/process.c++

```diff
@@ -137,4 +139,56 @@ void ProcessModule::exitImpl(jsg::Lock& js, int code) {
   js.logWarning(kj::str(err.get(js, "stack"_kj)));
 }
 
+kj::String ProcessModule::getCwd(jsg::Lock& js) {
+  auto cwd = getCurrentWorkingDirectory();
+  if (cwd == nullptr) {
+    return kj::str("/");
+  }
+  return kj::str("/", cwd.toString());
+}
+
+void ProcessModule::setCwd(jsg::Lock& js, kj::String path) {
+  static constexpr size_t MAX_PATH_LENGTH = 4096;
+  if (path.size() > MAX_PATH_LENGTH) {
+    throwUVException(js, UV_ENAMETOOLONG, "chdir"_kj);
+  }
+
+  if (path.size() == 0) {
+    throwUVException(js, UV_ENOENT, "chdir"_kj);
+  }
+
+  auto& vfs = VirtualFileSystem::current(js);
+  auto url = KJ_REQUIRE_NONNULL(jsg::Url::tryParse(path, "file:///"_kj));
+
+  // Resolve the path against current working directory if it's relative
+  kj::Path resolvedPath = [&]() {
```

Comment:
You're parsing the path above into a url, then again as a path here. The below here should probably work off the url's `pathname` instead for consistency with the rest of the fs apis.

Reply:
Thanks, that was from a previous iteration where I was doing the URL approach, this line just got left behind.